netty/src/main/java/io/grpc/netty/Utils.java

```diff
@@ -168,26 +169,33 @@ private static Http2Headers convertMetadata(Metadata headers) {
   }
 
   public static Status statusFromThrowable(Throwable t) {
+    // First, inspect the causal chain explicitly looking for status exceptions.
     Status s = Status.fromThrowable(t);
     if (s.getCode() != Status.Code.UNKNOWN) {
+      // It was already a status exception, just use it.
       return s;
     }
-    // TODO(ejona): reenable once startup races are resolved; ClosedChannelException is being seen
-    // still. Some tests are asserting UNAVAILABLE and were "working" previously but are now
-    // detecting that our behavior is flaky. See #1330
-    if (false && t instanceof ClosedChannelException) {
-      // ClosedChannelException is used any time the Netty channel is closed. Proper error
-      // processing requires remembering the error that occurred before this one and using it
-      // instead.
-      return Status.UNKNOWN.withCause(t);
-    }
-    if (t instanceof IOException) {
-      return Status.UNAVAILABLE.withCause(t);
-    }
-    if (t instanceof Http2Exception) {
-      return Status.INTERNAL.withCause(t);
+
+    // Assume UNKNOWN unless we find a better status.
+    String msg = t.getMessage();
```

Comment:
There needs to be some message, since the stack trace won't make it over the wire.